buildconfig/CMake/CommonSetup.cmake

```diff
@@ -412,6 +412,23 @@ if(NOT BUNDLES)
   set(BUNDLES "./")
 endif()
 
+# ##############################################################################
+# Setup pre-commit here as otherwise it will be overwritten by earlier 
+# pre-commit hooks being added
+# ##############################################################################
+if (MSVC)
```

Comment:
Please put this inside an `ENABLE_PRECOMMIT` which can be setup with a line before it being
```cmake
option(ENABLE_PRECOMMIT "Enable pre-commit framework" ON)
```

Reply:
Done :+1: